crates/cxx-qt-gen/src/parser/qenum.rs

```diff
@@ -0,0 +1,162 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use quote::ToTokens;
+use syn::{Ident, ItemEnum, Result, Variant};
+
+use crate::syntax::path::path_compare_str;
+
+pub struct ParsedQEnum {
+    /// The ident of the QEnum
+    pub ident: Ident,
+    /// the values of the QEnum
+    pub variants: Vec<Ident>,
+    /// The original enum item
+    pub item: ItemEnum,
+}
+
+impl ParsedQEnum {
+    fn parse_variant(variant: &Variant) -> Result<Ident> {
+        fn err(spanned: &impl ToTokens, message: &str) -> Result<Ident> {
```

Comment:
what the reason for this being a `fn` rather than a closure ? like `let err = |spanned: T, message: T| -> T { ... }` ?

Reply:
You can't do `&impl ToTokens` in a closure unfortunately, as that's using generics.

Reply:
Aha thought there was a reason, shame :-/